### ISSUE TITLE
Limited to 1 Context Menu Due to Hard-Coded ID

### ISSUE THREAD

**Issue by bogan27** [amurp/react-context-menu#11]
Because the ID is hardcoded to `contextMenu`, there can only be one context menu in the DOM at a time. Sometimes, you will want to have multiple menus with different options defined.

**Comment by amurp** [comment]
Good point. I'll see about adding an optional "id" prop to <ContextMenu />, which can default to the old "contextMenu" if undefined. It'll probably be a time to make the render method more smart and only render when opened, so you don't end up with a bunch of menus in the DOM at once when using multiple ContextMenus.

**Comment by bogan27** [comment]
Hi @amurp-  I wanted to use this in a project I'm currently working on so have this done locally already (the dynamic ID as well as the "smart" rendering to only have 1 menu open at a time). Also added in class names for style overrides. Just want to test it locally a little more, then I'll submit a PR for it within the next day or so.

**Comment by bogan27** [comment]
Just submitted https://github.com/amurp/react-context-menu/pull/12
